src/commonTest/kotlin/it/krzeminski/snakeyaml/engine/kmp/issues/issue54/DumpWithoutSpaceTest.kt

```diff
@@ -0,0 +1,64 @@
+package it.krzeminski.snakeyaml.engine.kmp.issues.issue54
+
+import io.kotest.assertions.fail
+import io.kotest.core.spec.style.FunSpec
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.shouldNotBe
+import io.kotest.matchers.string.shouldContain
+import it.krzeminski.snakeyaml.engine.kmp.api.Dump
+import it.krzeminski.snakeyaml.engine.kmp.api.DumpSettings
+import it.krzeminski.snakeyaml.engine.kmp.api.Load
+import it.krzeminski.snakeyaml.engine.kmp.api.LoadSettings
+
+/**
+ * Issue 54: add a space after anchor (when it is a simple key)
+ */
+class DumpWithoutSpaceTest : FunSpec({
+    fun parse(data: String): Any? {
+        val loadSettings = LoadSettings.builder().setAllowRecursiveKeys(true).build();
+        val load = Load(loadSettings);
+        return load.loadOne(data);
+    }
+
+    test("The document does not have a space after the *1 alias") {
+        try {
+            parse(
+                """|--- &1
+                   |hash:
+                   |  :one: true
+                   |  :two: true
+                   |  *1: true""".trimMargin()
+            )
+            fail("Shouldn't reach here!")
+        } catch (e: Exception) {
+            e.message shouldContain "could not find expected ':'"
+        }
+    }
+
+    test("The output does include a space after the *1 alias") {
+        val obj = parse(
+            """|--- &1
+               |hash:
+               |  :one: true
+               |  :two: true
+               |  *1 : true""".trimMargin()
+        )
+        obj shouldNotBe null
```

Comment:
There's some kind of an infinite loop here which makes the test time out. Digging deeper.

Reply:
Commenting out this assertion makes the tests pass. Let's investigate why the assertion fails, and if obj is really null here.

Reply:
It turned out it's enough to replace it with `obj.shouldNotBeNull()`.